### PR TITLE
Bootstrap + Catalog

### PR DESCRIPTION
**Pull request by eriknelson**
* Implements `broker::Bootstrap` for loading specs from a registry and persisting them
* Simplifies broker organization until we need to start splitting things off
* Added Spec -> Service transform
* Added `Description` to spec, the last missing required field (aside from plans, which is still hacked in)
* Implements spec conformant `broker::Catalog`

### CODE REVIEW

**Comment by eriknelson** [comment]
@jmrodri planning on following this up with a PR to close issues that make sense with codeclimate. Also want to talk to you about what issue types we should ignore (like TODO) so that we aren't getting false Xs on PRs. Maybe we nix the codeclimate integration all together from PRs...

File: cmd/mock-registry/ansibleapps.yaml [hunk 1]
```diff
@@ -2,17 +2,21 @@
...
   - id: 507856c7-8e08-460c-a0b2-7d8af8c4318d
     name: fusor/drupal-ansibleapp
+    description: Another website thing
     bindable: false
     async: "required"
   - id: f819ddd5-37d1-4698-b3fb-a3cc99a35d2e
     name: fusor/ghost-ansibleapp
+    description: Blogging and stuff
     bindable: false
     async: "unsupported"
```
Comment (jmrodri) [inline cmd/mock-registry/ansibleapps.yaml]: +1 to description.

File: pkg/ansibleapp/types.go [hunk 2]
```diff
@@ -4,10 +4,13 @@ import (
 	"encoding/json"
 )
 
+type SpecManifest map[string]*Spec
+
 type Spec struct {
-	Id       string `json:"id"`
-	Name     string `json:"name"`
-	Bindable bool   `json:"bindable"`
+	Id          string `json:"id"`
```
Comment (jmrodri) [inline pkg/ansibleapp/types.go]: I'm ok with Id being Id.

File: pkg/ansibleapp/types_test.go [hunk 3]
```diff
@@ -12,33 +12,36 @@ const SPEC_ID = "ab094014-b740-495e-b178-946d5aa97ebf"
 const SPEC_NAME = "fusor/etherpad-ansibleapp"
 const SPEC_BINDABLE = false
 const SPEC_ASYNC = "optional"
+const SPEC_DESCRIPTION = "A note taking webapp"
```
Comment (jmrodri) [inline pkg/ansibleapp/types_test.go]: FIX: don't use all caps

File: pkg/app/app.go [hunk 4]
```diff
@@ -3,15 +3,19 @@ package app
 import (
 	"fmt"
 	"github.com/fusor/ansible-service-broker/pkg/ansibleapp"
+	"github.com/fusor/ansible-service-broker/pkg/broker"
 	"github.com/fusor/ansible-service-broker/pkg/dao"
+	//"github.com/fusor/ansible-service-broker/pkg/handler"
+	//"net/http"
```
Comment (jmrodri) [inline pkg/app/app.go]: I assume these commented out imports are for debugging?

Comment (eriknelson) [inline pkg/app/app.go]: Yeah, I left these in along with the calls in `App::Start()` to just indicate the intention here and to allow for it to easily be turned on for testing purposes. Long term, both the imports are necessary.

**Pull request merged**